api/clients/v2/relay_client.go

```diff
@@ -52,6 +54,20 @@ type RelayClient interface {
 	Close() error
 }
 
+// grpcClient encapsulates a gRPC client and a channel for limiting concurrent requests to that client.
+type grpcClient struct {
+	// client is the gRPC client used for sending gRPC requests.
+	client relaygrpc.RelayClient
```

Comment:
Style question:

Did you consider embedding the `relaygrpc.RelayClient` in this new struct? Doing so avoids the awkward `client.client` syntax brought on by the current setup

Reply:
I've refactored this... the concurrency limiting is now completely encapsulated inside the struct `limitedClient`.